File: bench/tasks/06-where-sum/go/main.go
package main

import "fmt"

func main() {
	numbers := []int{4, 7, 2, 9, 12}
	var large []int
	for _, n := range numbers {
		if n > 5 {
			large = append(large, n)
		}
	}
	total := 0
	for _, n := range large {
		total += n
	}
	fmt.Println("Sum: " + fmt.Sprint(total))
	if total != 28 {
		panic("Unexpected collection total.")
	}
}

File: bench/tasks/06-where-sum/rust/main.rs
fn main() {
    let numbers = [4, 7, 2, 9, 12];
    let total: i32 = numbers.iter().copied().filter(|&n| n > 5).sum();
    println!("Sum: {}", total);
    assert_eq!(total, 28, "Unexpected collection total.");
}
